src/types/from_register_input.rs

```diff
@@ -1,13 +1,94 @@
 use std::time::Duration;
 
+use bytes::{BufMut, Bytes, BytesMut};
 use compositor_pipeline::{
-    pipeline::{self, decoder, input},
+    pipeline::{
+        self, decoder,
+        input::{self, rtp::InputAudioStream},
+    },
     queue,
 };
 
 use super::register_input::*;
 use super::util::*;
 
+/// [RFC 3640, section 4.1. MIME Type Registration (`config` subsection)](https://datatracker.ietf.org/doc/html/rfc3640#section-4.1)
+fn parse_asc(asc: Option<&str>) -> Result<Option<Bytes>, TypeError> {
+    let Some(asc) = asc else {
+        return Ok(None);
+    };
+
+    const EMPTY_ASC: &str = "The AudioSpecificConfig field is empty. Either omit this field or provide a non-empty ASC.";
+    if asc.is_empty() {
+        return Err(TypeError::new(EMPTY_ASC));
+    }
+
+    let mut output = BytesMut::new();
+
+    const BAD_ASC_FORMAT: &str = "Not all of the provided AudioSpecificConfig is hex digits.";
+    if !asc.chars().all(|c| char::is_ascii_hexdigit(&c)) {
+        return Err(TypeError::new(BAD_ASC_FORMAT));
+    }
+
+    for a in asc.as_bytes().chunks(2) {
```

Comment:
If you modify container from the loop then try to declare that mut variable as close as possible.

Also in this specific case I would consider functional approach

```
    let output = asc.as_bytes().chunks(2).map(|chunk|{
        ...
        let parsed = match u8::from_str_radix(&val, 16) {
            Ok(res) => res,
            // This can't happen, but just to be safe
            Err(_) => Err(TypeError::new(BYTE_PARSE_ERROR)),
        };
    })
    .collect()?;
```

Reply:
```suggestion
    for chunk in asc.as_bytes().chunks(2) {
```

This is not Go :D